test/xpu_api/numerics/complex.number/cases.h

```diff
@@ -218,48 +215,51 @@ CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
         return non_zero_nan;
     }
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
 inline
 int
 classify(double x)
 {
-// Suppress clang warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
-
     if (x == 0)
         return zero;
     if (std::isinf(x))
         return inf;
     if (std::isnan(x))
         return NaN;
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
-void is_about(float x, float y)
+auto
+get_eps(float)
 {
-// Suppress clang warning: floating-point comparison is always true; constant cannot be represented exactly in type 'float' [-Wliteral-range]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_LITERAL_RANGE
-
-    assert(std::abs((x-y)/(x+y)) < 1.e-6);
+    constexpr float eps = 1.e-6;
+    return eps;
+};
 
-CLANG_DIAGNOSTIC_POP
-}
+template <typename T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, T>::type
+get_eps(T)
+{
+    static_assert(::std::is_same_v<T, double> || ::std::is_same_v<T, long double>);
+    constexpr T eps = 1.e-11;
+    return eps;
+};
 
-void is_about(double x, double y)
+template <typename T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, void>::type
+is_about(T x, T y)
 {
-    assert(std::abs((x-y)/(x+y)) < 1.e-14);
+    const auto exp = get_eps(T{});
```

Comment:
Can we rely on `std::numeric_limits<T>::epsilon()`, with some correction, if it needs?
(it is available starting from C++17).

Or if not, we can do such customization in simpler way
`is_about(T x, T y, T e = get_eps<T>::val)
{...}`
Where get_eps is:
```
template<typename> struct get_eps;
template<> struct get_eps<float> { static constexpr float val = 1e-6;};
template<> struct get_eps<double> { static constexpr double val = 1e-16;};
```
https://godbolt.org/z/bKWbhs7Yq

Reply:
About  ```std::numeric_limits<T>::epsilon()``` - it's too small for our check.